features/telemetry/protocols/bmp/adapter/src/main/java/org/opennms/netmgt/telemetry/protocols/bmp/adapter/openbmp/BmpKafkaProducer.java

```diff
@@ -59,7 +60,12 @@ public BmpKafkaProducer(final String topicPrefix,
     }
 
     private static KafkaProducer<String, String> buildProducer(final Map<String, Object> kafkaConfig) {
-        // TODO fooker: Apply defaults (steal from https://github.com/SNAS/openbmp/blob/1a615a3c75a0143cc87ec70458471f0af67d3929/Server/src/kafka/MsgBusImpl_kafka.cpp#L162) (see https://issues.opennms.org/browse/NMS-12574)
+        // Apply some defaults
+        kafkaConfig.putIfAbsent(ProducerConfig.BATCH_SIZE_CONFIG, 100);
+        kafkaConfig.putIfAbsent(ProducerConfig.COMPRESSION_TYPE_CONFIG, "snappy");
```

Comment:
I think these two parameters are sufficient for increasing the default message size. By default it is 1MB, you may want to increase it to 10MB.
 `kafkaConfig.putIfAbsent(ProducerConfig.COMPRESSION_TYPE_CONFIG, "snappy");`
 `kafkaConfig.putIfAbsent(ProducerConfig.MAX_REQUEST_SIZE_CONFIG, 1000000);`

Also we should allow the config to be added from osgi configuration just like [here](https://github.com/OpenNMS/opennms/blob/3de338a7044ed5d45df5b70f26fda5798530e8e5/core/ipc/common/kafka/src/main/java/org/opennms/core/ipc/common/kafka/OsgiKafkaConfigProvider.java#L59)

The loaded config should override anything that we set here.  So we can totally skip these defaults and allow user to add them.

Reply:
We don't use our default kafka config mechanism here, because we talk to the external OpenBMP kafka cluster. Therefore the config is only set in the according adapter. And this config is already overriding the defaults, right?

Reply:
okay, looks like it is set in adapter config. That works.